include/clap/ext/draft/channel-info.h

```diff
@@ -0,0 +1,61 @@
+#pragma once
+
+#include "../../plugin.h"
+#include "../../color.h"
+#include "../../string-sizes.h"
+
+static CLAP_CONSTEXPR const char CLAP_EXT_CHANNEL_INFO[] = "clap.channel-info.draft/0";
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+enum {
+   CLAP_CHANNEL_INFO_HAS_TRACK_NAME = (1 << 0),
+   CLAP_CHANNEL_INFO_HAS_TRACK_COLOR = (1 << 1),
+   CLAP_CHANNEL_INFO_HAS_TRACK_ID = (1 << 1),
+   CLAP_CHANNEL_INFO_HAS_TRACK_PATH = (1 << 1),
+   CLAP_CHANNEL_INFO_HAS_CHANNEL_NAME = (1 << 2),
+   CLAP_CHANNEL_INFO_HAS_CHANNEL_COLOR = (1 << 3),
+   CLAP_CHANNEL_INFO_HAS_CHANNEL_ID = (1 << 4),
+   CLAP_CHANNEL_INFO_HAS_CHANNEL_PATH = (1 << 4),
+   CLAP_CHANNEL_INFO_HAS_AUDIO_CHANNEL_COUNT = (1 << 5),
+   CLAP_CHANNEL_INFO_HAS_AUDIO_PORT_TYPE = (1 << 6),
+};
+
+typedef struct clap_channel_info {
+   uint32_t flags; // see the flags above
+
+   // info relatives to the track to which the current channel belong
+   char         track_id[CLAP_CUSTOM_ID_SIZE];
+   char         track_name[CLAP_NAME_SIZE];
+   char         track_path[CLAP_PATH_SIZE];
+   clap_color_t track_color;
+   bool         is_return_track;
```

Comment:
Before this is merged, every field needs a very clear explanation of what the expected values are. And also mention things like, if both the track and channel path are set, then the track path must be a prefix of the channel path. And I'm not sure if 'channel' is the clearest name for this. Not sure what a better alternative would be though. Sub-track? 'Plugin path' also made sense to me, except for the part where the path points not to the plugin but to the thing holding the plugin so that's also not a good choice.

Reply:
Maybe I should rename it to `device_chain_info`.